### PR TITLE
Fix test for categorical sampling with multi-D logits

### PR DESCRIPTION
**Pull request by juliuskunze**
Fixes test code of `random.categorical` for the case of multi-dimensional logits. This faulty test case was earlier unearthed and disabled after a fix in the chi-squared test implementation, as described in https://github.com/google/jax/pull/2847.

### CODE REVIEW

**Comment by shoyer** [comment]
Thanks @JuliusKunze !
